src/Sarif/ExtensionMethods.cs

```diff
@@ -562,6 +562,36 @@ internal static string PropertyValue(this Dictionary<string, string> properties,
             }
         }
 
+        /// <summary>
+        /// Extension method for comparing nullable structs
+        /// with proper handling of the case when 'left' is null.
+        /// </summary>
+        /// <typeparam name="T">The type of struct.</typeparam>
+        /// <param name="left">The first item to compare.</param>
+        /// <param name="right">The second item to compare.</param>
+        /// <returns>
+        /// Returns 1 if left is greater than right, -1 if left is less than right, 0 if they are equal.
+        /// </returns>
+        internal static int CompareTo<T>(this T? left, T? right) where T : struct, IComparable
```

Comment:
Did we just implement Nullable.Compare<Guid> here? :)

I think so. If I'm correct, let's zap this helper and use the C# built-in.

[https://learn.microsoft.com/en-us/dotnet/api/system.nullable.compare?view=net-6.0](https://learn.microsoft.com/en-us/dotnet/api/system.nullable.compare?view=net-6.0)

Reply:
Good catch. I have just updated the extension method to use the C# built-in. Let me close this one and reply on the other so no need to look duplicated